session.go

```diff
@@ -5212,3 +5226,67 @@ func hasErrMsg(d []byte) bool {
 	}
 	return false
 }
+
+// getRFC2253NameString converts from an ASN.1 structured representation of the certificate
+// to a UTF-8 string representation(RDN) and returns it.
+func getRFC2253NameString(certificate *x509.Certificate) string {
```

Comment:
This should have unit tests covering at the very least, the examples in section 5 of the [RFC](https://www.ietf.org/rfc/rfc2253.txt) - if you make a function that accepts the unmarshaled `pkix.RDNSequence` and does the actual work you should be able to unit test by the logic of this func.

If you do the above, `getRFC2253NameString()` becomes a helper func accepting a `x509.Certificate` that calls the actual logic func - maybe the helper func can be `getRFC2253NameStringFromCert()` that calls `getRFC2253NameString()` or something - up to you.